test/parallel/test-frozen-global.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+// This tests that the globals can still be loaded after globalThis is freezed.
+
+require('../common');
+const assert = require('assert');
+
+Object.freeze(globalThis);
+const keys = Reflect.ownKeys(globalThis).filter((k) => typeof k === 'string');
+
+// These failures come from undici. We can remove them once they are fixed.
+const knownFailures = new Set(['FormData', 'Headers', 'Request', 'Response']);
+const failures = new Map();
+const success = new Map();
+for (const key of keys) {
+  try {
+    success.set(key, globalThis[key]);
+  } catch (e) {
+    failures.set(key, e);
+  }
```

Comment:
```suggestion
  const access = () => globalThis[key];
  if (knownFailures.has(key)) {
    assert.throws(access, { name: 'TypeError' });
    failures.delete(key);
  } else {
    access();
  }
```

Reply:
Why is it deleting the failure (which is undeclared if we apply the suggestion above)?

Reply:
Sorry I meant 

```suggestion
  const access = () => globalThis[key];
  if (knownFailures.has(key)) {
    assert.throws(access, { name: 'TypeError' });
    knownFailures.delete(key);
  } else {
    access();
  }
```

Reply:
I think this version would lead to less debuggable messages - for example if the error is caused by redefining something else on a global in the implementation of another (e.g. the undici symbol for the known failures listed here), the error message would not show which key is causing the trouble, and only shows the symbol which is an implementation detail of that key. So fixing the known failures takes 4 trial-and-errors. Whereas having a failures map for them all and logging the map only takes 1 trail-and-error.

Reply:
What do you think of the following then?

```suggestion
  const access = () => globalThis[key];
  if (knownFailures.has(key)) {
    assert.throws(access, { name: 'TypeError' });
    knownFailures.delete(key);
  } else {
    try { access(); } catch { unexpectedFailures.add(key); }
  }
```

Reply:
(And also I personally prefer to avoid one-line helpers like `access()` here in the test because they make the test failures harder to read. IMO tests are usually easier to understand if we can just be verbose and write the code being tested directly even if it involves some repetition - as long as it's not too much repetition)

Reply:
At least you can agree that the `success` map is not useful, nor for the test itself, and likely not for debugging either?

Reply:
> What do you think of the following then?

I think this is more complicated than the original version, which just adds results on success and error on failure to two maps instead of trying to add or delete anything? At least it'd take me a while to understand what the second version is doing, whereas the original one doesn't take much brain power to understand.

Reply:
> At least you can agree that the success map is not useful, nor for the test itself, and likely not for debugging either?

That avoids having to add a comment to skip the no-unused-expressions eslint check. I figure if we have to throw extra code to avoid it, might as well just put them into a map, in case one wants to find out what can actually be loaded (for example, to find out `fetch` can be loaded, but other fetch-related classes can't).